pkg/controller/certificaterequests/sync.go

```diff
@@ -158,14 +158,14 @@ func (c *Controller) updateCertificateRequestStatusAndAnnotations(ctx context.Co
 	log := logf.FromContext(ctx, "updateStatus")
 
 	// if annotations changed we have to call .Update() and not .UpdateStatus()
-	if !reflect.DeepEqual(old.Annotations, new.Annotations) {
+	if !apiequality.Semantic.DeepEqual(old.Annotations, new.Annotations) {
```

Comment:
For annotations I think `reflect.DeepEqual` is better since here we would want to catch any update, so perhaps no need to change it here.

Reply:
Okay, I have reverted it accordingly